extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -252,9 +252,38 @@ function createDicomWebApi(dicomWebConfig, UserAuthenticationService) {
         sortFunction
       );
 
+      /**
+       * naturalizes the dataset, and adds a retrieve bulkdata method
+       * to any values containing BulkDataURI.
+       * @param {*} instance
+       * @returns naturalized dataset, with retrieveBulkData methods
+       */
+      const addRetrieveBulkData = instance => {
+        const naturalized = naturalizeDataset(instance);
+        Object.keys(naturalized).forEach(key => {
+          const value = naturalized[key];
+          if (value && value.BulkDataURI && !value.Value) {
```

Comment:
what are you checking for `!valu.Value`? are you checking for null/undefined, can it be 0?

Reply:
mabye we add a comment above this if statement to say what is the branch here

Reply:
value.Value can only be a binary byte array for pixel data, so, no, 0 isn't an option.  The reason this is checked here is that
the value might already have been read, in which case it isn't necessary to read it a second time.

Reply:
Updated with a comment.

Reply:
@wayfarer3130  Something that just came to my mind is that how many fields have BulDataURI, since wee are checking for all the dicom keys and all the instances, and it might be slow ....

Reply:
It is a lot faster than always fetching all the data, and I suspect is faster than parsing the data once (which can take a noticeable amount of time).   
It is also currently only done on the top level tags.  It would be possible to have a set of tag values to check, but given how overlays are encoded, I suspect that would be slower because there isn't a good way to iterate over just those.
I don't think this is a significant performance concern right now.

Reply:
I see, thanks.